vips/foreign.c

```diff
@@ -1,29 +1,92 @@
 #include "lang.h"
 #include "foreign.h"
+#include "govips.h"
 
-int load_image_buffer(void *buf, size_t len, int imageType, VipsImage **out) {
+int load_image_buffer_operation(VipsOperation *operation, VipsBlob *blob, VipsImage **out, ImportParams *params) {
+    int result;
+
+    // Set operation parameters
+    result = vips_object_set(VIPS_OBJECT(operation), "buffer", blob, "out", out, NULL);
+    if (result) {
+        return result;
+    }
+
+    if (params->fail_set) {
+        result = vips_object_set(VIPS_OBJECT(operation), "fail", params->fail, NULL);
+        if (result) {
+            govipsLoggingHandler("govips", G_LOG_LEVEL_INFO, "can't set optional parameter \"fail\"");
```

Comment:
Are you sure we want to carry on if this fails? In what cases does it fail? If I load an image with autorotate, and it fails to carry it out, this would just log and carry on without the user code being able to deal w/ it. Why not just fail altogethre?

Reply:
As I understand it, the fail flag is not supported only if partial image loading is supported. In this case, if it is not supported, then it can simply be ignored.
In such a situation, loading images will behave as if this flag is true.